### DISCUSSION TITLE
Built-in function to detect the kind of a request (like Audio, Video, Image, MIDI, Stream, …)?

### DISCUSSION THREAD

**Discussion by Moonbase59** [savonet/liquidsoap#3839] [Q&A]
For use in the `autocue2` protocol (or in general), do we have a builtin function to say what we got in a request or file?

I’d like to start processing on _audio_ and _video_ content _only_, **not** on images, MIDI, remote streams or other types. (See also: https://github.com/Moonbase59/autocue/issues/1)

If that doesn’t exist, can we somehow use the `kind` metadata returned when resolving a request, and is this always available?

**Comment by toots** [comment]
That's a good idea. I'll try to implement something. Meanwhile, you can access the file that was resolved and inspect it via `ffprobe` or similar.
